components/Iterations.tsx

```diff
@@ -0,0 +1,78 @@
+import { useEffect } from 'react';
+import { ButtonDropdown } from '@geist-ui/react';
+import { parseCookies } from 'nookies';
+import { useSelector, useDispatch } from 'react-redux';
+import { State, Iteration } from '../redux/types';
+import { addIterations } from '../redux/actions/iterations.actions';
+import { parseISO, format } from 'date-fns';
+
+interface IterationsParams {
+  id: string;
+  selectedIteration?: Iteration;
+  addIteration: (name: string, filter: Iteration) => void;
+  removeIteration: (name: string, filter: Iteration) => void;
+}
+
+const Iterations = ({ id, selectedIteration, addIteration, removeIteration }: IterationsParams): JSX.Element => {
+  const { apiToken } = parseCookies(); //todo: move to redux
+  const iterations = useSelector((state: State): Iteration[] => {
+    const iterations = state.iterations[id] || new Set();
+    return Array.from(iterations);
+  });
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (!iterations.length && Boolean(id)) {
+      //todo: move this out to an action
+      const getIterations = async () => {
+        const result = await fetch(
+          `https://www.pivotaltracker.com/services/v5/projects/${id}/iterations?limit=20&scope=current_backlog`,
+          {
+            headers: {
+              'X-TrackerToken': apiToken,
+            },
+          }
+        );
+
```

Comment:
What do you mean by moving this out to an action? are you referring to the network request

Reply:
Yeah basically, I don't think the network requests should belong inside the components, I need to refactor them somewhere else in the action model